projects/novo-elements/src/elements/data-table/services/static-data-table.service.ts

```diff
@@ -28,9 +28,19 @@ export class StaticDataTableService<T> implements IDataTableService<T> {
         total = this.currentData.length;
       }
       if (filter) {
-        let value = Helpers.isString(filter.value) ? filter.value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&') : filter.value;
-        this.currentData = this.currentData.filter(Helpers.filterByField(filter.id, value));
-        total = this.currentData.length;
+        let filters = Helpers.convertToArray(filter);
+        filters.forEach((aFilter) => {
+          if (Array.isArray(aFilter.value)) {
+            let values = Helpers.convertToArray(aFilter.value).map((aValue) => {
+              return Helpers.cleanIfString(aValue);
+            });
+            this.currentData = this.currentData.filter(Helpers.filterByField(aFilter.id, values));
+          } else {
+            let value = Helpers.cleanIfString(aFilter.value);
+            this.currentData = this.currentData.filter(Helpers.filterByField(aFilter.id, value));
+          }
```

Comment:
can this logic be moved out to its own function? that way, it can be tested separately too.